tests/test_integration_http.py

```diff
@@ -575,6 +574,8 @@ def test_transaction_only(cmake, httpserver, build_args):
     timestamp = time.strptime(payload["timestamp"], RFC3339_FORMAT)
     assert timestamp >= start_timestamp
 
+    assert trace_context["data"] == {"url": "https://example.com"}
```

Comment:
Moved the expected data from the `transaction_data` check into this `trace_context.data` check; should we now also make sure that the "extra" data in the transaction is cleared up at this point (since we remove the value when we store it in the trace context data)?

Reply:
Nope, since this was a workaround (adopted from the workarounds of SDKs), it is unnecessary to encode that it is missing. 

But I think it would make sense to rename the field from "extra" back to "data", since naming it "extra" was the workaround. We're not breaking anything by changing this back to "data". Asserting that "data" is not an existing key in the transaction root can make sense.

Reply:
To be clear, starting from the client interface. We now do the following:

```
sentry_transaction_set_data() -> 
sets tx["extra"] -> 
sentry_transaction_finish() -> 
moves tx["extra"] to tx["contexts.trace.data"]
```

This means that the interface names it "data," and then we name it "extra" until the transaction finishes. At this point, we name it "data" again and move to another child object.

That will be indecipherable at some point. We should change the intermediate "extra" back to "data" so that the change only concerns moving it from the transaction root to "contexts.trace".

Reply:
Changed the "extra" key to "data"; would we now want to assert + warn when the "data" key already exists in the [`set_data` function](https://github.com/getsentry/sentry-native/blob/0623fe6224f55e42ed8c107bedeff90b82c69186/src/sentry_tracing.c#L559-L570)